tests/taskgraphs/test_depgraph.py

```diff
@@ -0,0 +1,29 @@
+import unittest
+
+from tiledb.cloud.taskgraphs import depgraph
+
+
+class TestDepGraph(unittest.TestCase):
```

Comment:
please add test cases for:

* empty graph
* single-node graph

things like this can surface otherwise-non-obvious corner-case issues

even if your code correctly handles these cases today, in your python-expert hands, it might not continue to do so in the future. let's guard-rail it a bit here.

Reply:
Added tests for these cases.